program/src/state.rs

```diff
@@ -1647,7 +1649,7 @@ impl MangoAccount {
                     0,
                     fill.maker_timestamp,
                     fill.timestamp,
-                    fill.quantity,
+                    fill.quantity.checked_shl(lm_qty_shift as u32).unwrap(),
```

Comment:
Hmm, are you sure this will work? The quantity will be limited by the `max_depth_size`.

I think it may work better with the incentive function to reduce the quantity when rewarding cancels?

Also, how about using a mul instead of a shift?

Reply:
This is a good point.

Reply:
One side effect is this will reward you more the longer you were on the book. I think fills and cancels should actually be treated separately. Or perhaps, a fill can be given plus 100 seconds?